File: PubSubDemo/WebDemo/common/PubSub/ts/Messages.ts
module WebDemo {

    export interface ISubscribe extends IMessage {
        Topic: string;
    }

    export class Subscribe extends Message implements ISubscribe {
        Topic: string;
    }

    export interface IUnsubscribe extends IMessage {
        Topic: string;
    }

    export class Unsubscribe extends Message implements IUnsubscribe {
        Topic: string;
    }

    export interface ITopicUpdate extends IMessage {
        Publisher: string;
        Date: Date;
        Topic: string;
        Content: string;
    }

    export class TopicUpdate extends Message implements ITopicUpdate {
        Publisher: string;
        Date: Date;
        Topic: string;
        Content: string;
    }

    export interface IGetSubscriptions extends IMessage {
    }

    export class GetSubscriptions extends Message implements IGetSubscriptions {
    }

    export interface IGetTopics extends IMessage {
    }

    export class GetTopics extends Message implements IGetTopics  {
    }

    export interface ISubscriptions extends IMessage {
        List: string[];
    }

    export class Subscriptions extends Message implements ISubscriptions {
        List: string[];
    }

    export interface ITopics extends IMessage {
        List: string[];
    }

    export class Topics extends Message implements ITopics {
        List: string[];
    }

}

File: PubSubDemo/WebDemo/common/PubSub/cs/Messages.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using Messages;

namespace PubSub
{
    public sealed class Subscribe : Message
    {
        public string Topic { get; set; }

        public Subscribe() { }

        public Subscribe(string topic) : this()
        {
            this.Topic = topic;
        }
    }

    public sealed class Unsubscribe : Message
    {
        public string Topic { get; set; }

        public Unsubscribe() { }

        public Unsubscribe(string topic) : this()
        {
            this.Topic = topic;
        }
    }

    public sealed class TopicUpdate : Message
    {
        public string Publisher { get; set; }
        public DateTime Date { get; set; }
        public string Topic { get; set; }
        public string Content { get; set; }

        public TopicUpdate() { }

        public TopicUpdate(string topic, string content)
        {
            this.Topic = topic;
            this.Content = content;
        }
    }

    public sealed class Error : Message
    {
        public string Message { get; set; }

        public Error() { }

        public Error(string e) : this()
        {
            this.Message = e;
        }
    }

    public sealed class GetSubscriptions : Message
    {
        public GetSubscriptions() { }
    }

    public sealed class GetTopics : Message
    {
        public GetTopics() { }
    }

    public sealed class Subscriptions : Message
    {
        public List<string> List { get; set; }
        public Subscriptions() { }

        public Subscriptions(List<string> list)
        {
            this.List = list;
        }
    }

    public sealed class Topics : Message
    {
        public List<string> List { get; set; }
        public Topics() { }

        public Topics(List<string> list)
        {
            this.List = list;
        }
    }
}
